packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -222,21 +236,96 @@ async function selectPlatformAndExecute(
     const message = isAndroid
       ? 'force_lightning_lwc_android_target_remembered'
       : 'force_lightning_lwc_ios_target_remembered';
-    placeholderText = nls.localize(message, lastTarget);
+    createDevicePlaceholderText = nls.localize(message, lastTarget);
     target = lastTarget;
   }
-  const targetName = await vscode.window.showInputBox({
-    placeHolder: placeholderText
+
+  const deviceListOutput = new CommandOutput();
+  const deviceListCommand = new SfdxCommandBuilder()
+    .withArg(sfdxDeviceListCommand)
+    .withFlag('-p', platformSelection.platformName)
+    .withJson()
+    .build();
+
+  let deviceListExecutionExitCode: number | undefined;
+  const deviceListCancellationTokenSource = new vscode.CancellationTokenSource();
+  const deviceListCancellationToken = deviceListCancellationTokenSource.token;
+  const deviceListExecutor = new CliCommandExecutor(deviceListCommand, {});
+  const deviceListExecution = deviceListExecutor.execute(
+    deviceListCancellationToken
+  );
+  deviceListExecution.processExitSubject.subscribe(exitCode => {
+    deviceListExecutionExitCode = exitCode;
   });
 
-  if (targetName === undefined) {
-    vscode.window.showInformationMessage(
-      isAndroid
-        ? nls.localize('force_lightning_lwc_android_device_cancelled')
-        : nls.localize('force_lightning_lwc_ios_device_cancelled')
+  const options: vscode.QuickPickItem[] = [];
+  let targetName: string | undefined;
+
+  try {
+    const result: string = await deviceListOutput.getCmdResult(
+      deviceListExecution
     );
-    return;
+    const jsonString: string = result.substring(result.indexOf('{'));
+    const json: any = JSON.parse(jsonString);
+    const devices: any[] = json.result as any[];
```

Comment:
@maliroteh-sf Could you add actual definition for the interface that's used here: like `interface DeviceInformation {...}`. Ideally we don't want to use any `any`.